pkg/api/v1alpha1/cloudstackdatacenterconfig_types.go

```diff
@@ -142,6 +159,26 @@ func (v *CloudStackDatacenterConfig) Validate() error {
 	return nil
 }
 
+func (v *CloudStackDatacenterConfig) SetDefaults() {
+	if v.Spec.AvailabilityZones == nil || len(v.Spec.AvailabilityZones) == 0 {
```

Comment:
I think we need to add two extra validations:
* Make sure that v.Spec.Zones, v.Spec.Domain, v.Spec.Account,	v.Spec.ManagementApiEndpoint are all empty
* Make sure that `len(v.Spec.AvailabilityZones) > 0`

Reply:
I see. I think that makes sense to do in a follow-up PR once we migrate to actually using these types throughout. I can add a TODO here